cdxev/build_public_bom.py

```diff
@@ -1,14 +1,47 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 import json
+import logging
 import re
 import typing as t
 from pathlib import Path
-from typing import Sequence
+from typing import Any, Sequence
 
 from jsonschema import Draft7Validator, FormatChecker
 
 from cdxev.auxiliary.sbomFunctions import extract_components
+from cdxev.log import LogMessage
+
+logger = logging.getLogger(__name__)
+
+
+def check_affected_metadata(metadata: dict[str, Any], path_to_schema: Path) -> bool:
```

Comment:
Please use the already existing function, where it's checked for `components` instead of introducing a new function